tests/mpi/mesh_worker_01.cc

```diff
@@ -241,9 +241,9 @@ test()
 int
 main(int argc, char **argv)
 {
-  Utilities::MPI::MPI_InitFinalize mpi_initialization(
-    argc, argv, testing_max_num_threads());
-  MPILogInitAll log;
+  // Disable multithreading so that text output order is consistent
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+  MPILogInitAll                    log;
```

Comment:
Right now we don't support chunking in the TaskFlow implementation. This particular test (and the others changed) relies on the output being generated in a specific order as it compares line by line. I believe this order is not explicitly assured by the TBB implementation either but just happens to be consistent because this gets chunked to run sequentially with the default grain size in the TBB implementation. 

The outputs for the TaskFlow implementation are correct they just appear in a different order. I was advised that this would be the sensible change to make as right now I don't think this test actually even runs on more than 1 thread per MPI rank with the TBB implementation.

Reply:
I'm not sure that's true, but moreover I think that this essentially disables the test and that would be a shame. Can you show in which ways the output might differ depending on thread placement of tasks? Currently, the output of the whole test is this:
```
DEAL:0::DoFHandler ndofs=2
DEAL:0::*** 1. CELLS ***
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=0 faces_to_ghost=0
DEAL:0::* own_cells=1 ghost_cells=0 own_faces=0 faces_to_ghost=0
DEAL:0::C 0_0:
DEAL:0::* own_cells=0 ghost_cells=1 own_faces=0 faces_to_ghost=0
DEAL:0::C 1_0:
DEAL:0::* own_cells=1 ghost_cells=1 own_faces=0 faces_to_ghost=0
DEAL:0::C 0_0:
DEAL:0::C 1_0:
DEAL:0::*** 2. FACES ***
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=1 faces_to_ghost=0
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=2 faces_to_ghost=0
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=0 faces_to_ghost=1
DEAL:0::F cell1 = 0_0: face = 1 cell2 = 1_0: face2 = 0
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=0 faces_to_ghost=2
DEAL:0::F cell1 = 0_0: face = 1 cell2 = 1_0: face2 = 0
```
Which lines are unreliably ordered?

Reply:
Sorry to respond with the output of a different test but I just happen to have this one on hand. 

Here is the expected output which TBB generates for Mesh_Worker_02:
​
```
DEAL:0::DoFHandler ndofs=7
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=1 faces_to_ghost=0
DEAL:0::F cell1 = 0_1:1 face = 3 cell2 = 0_1:3 face2 = 2
DEAL:0::F cell1 = 0_1:2 face = 1 cell2 = 0_1:3 face2 = 0
DEAL:0::F cell1 = 0_2:00 face = 1 cell2 = 0_2:01 face2 = 0
DEAL:0::F cell1 = 0_2:00 face = 3 cell2 = 0_2:02 face2 = 2
DEAL:0::F cell1 = 0_2:01 face = 1 cell2 = 0_1:1 face2 = 0
DEAL:0::F cell1 = 0_2:01 face = 3 cell2 = 0_2:03 face2 = 2
DEAL:0::F cell1 = 0_2:02 face = 1 cell2 = 0_2:03 face2 = 0
DEAL:0::F cell1 = 0_2:02 face = 3 cell2 = 0_1:2 face2 = 2
DEAL:0::F cell1 = 0_2:03 face = 1 cell2 = 0_1:1 face2 = 0
DEAL:0::F cell1 = 0_2:03 face = 3 cell2 = 0_1:2 face2 = 2
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=0 faces_to_ghost=1
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=0 faces_to_ghost=2
```

And TaskFlow:

```
DEAL:0::DoFHandler ndofs=7
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=1 faces_to_ghost=0
DEAL:0::F cell1 = 0_1:1 face = 3 cell2 = 0_1:3 face2 = 2
DEAL:0::F cell1 = 0_1:2 face = 1 cell2 = 0_1:3 face2 = 0
DEAL:0::F cell1 = 0_2:01 face = 1 cell2 = 0_1:1 face2 = 0
DEAL:0::F cell1 = 0_2:00 face = 1 cell2 = 0_2:01 face2 = 0
DEAL:0::F cell1 = 0_2:01 face = 3 cell2 = 0_2:03 face2 = 2
DEAL:0::F cell1 = 0_2:00 face = 3 cell2 = 0_2:02 face2 = 2
DEAL:0::F cell1 = 0_2:03 face = 1 cell2 = 0_1:1 face2 = 0
DEAL:0::F cell1 = 0_2:02 face = 1 cell2 = 0_2:03 face2 = 0
DEAL:0::F cell1 = 0_2:03 face = 3 cell2 = 0_1:2 face2 = 2
DEAL:0::F cell1 = 0_2:02 face = 3 cell2 = 0_1:2 face2 = 2
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=0 faces_to_ghost=1
DEAL:0::* own_cells=0 ghost_cells=0 own_faces=0 faces_to_ghost=2
```

Reply:
basically, the test produce text output in the worker, that can be scheduled concurrently. The test used to work because the default chunk size with TBB is larger than the number of cells in this test. The proposed change make sure the test will run sequentially. 
I think this is okay because the purpose of the test is not parallel computing but that the mesh worker visits the correct cells and faces.

Reply:
I see, thank you!

Reply:
I think that the change here is still awkward because bypass all of the interesting stuff if we have no threads, which is what happens here. On the other hand, we do test `WorkStream` pretty heavily in many other tests, so that might not matter.